src/editor/controller/editor_controller.cpp

```diff
@@ -42,6 +42,7 @@
 #include "reports.hpp"
 
 #include "desktop/clipboard.hpp"
+#include "floating_label.hpp"
```

Comment:
Without it, this happens:

```
src/editor/controller/editor_controller.cpp: In member function ‘void editor::editor_controller::init_gui()’:
src/editor/controller/editor_controller.cpp:99:65: error: invalid use of incomplete type ‘struct font::floating_label_context’
  floating_label_manager_.reset(new font::floating_label_context());
                                                                 ^
In file included from src/editor/controller/editor_controller.cpp:23:0:
src/editor/controller/editor_controller.hpp:40:8: note: forward declaration of ‘struct font::floating_label_context’
 struct floating_label_context;
        ^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6/memory:81:0,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/date_time/time_clock.hpp:17,
                 from /usr/include/boost/date_time/posix_time/posix_time_types.hpp:10,
                 from src/log.hpp:59,
                 from src/editor/editor_common.hpp:25,
                 from src/editor/map/editor_map.hpp:18,
                 from src/editor/map/map_context.hpp:18,
                 from src/editor/map/context_manager.hpp:18,
                 from src/editor/controller/editor_controller.cpp:18:
/usr/include/c++/6/bits/unique_ptr.h: In instantiation of ‘void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = font::floating_label_context]’:
/usr/include/c++/6/bits/unique_ptr.h:236:17:   required from ‘std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = font::floating_label_context; _Dp = std::default_delete<font::floating_label_context>]’
/usr/include/c++/6/bits/unique_ptr.h:201:61:   required from ‘constexpr std::unique_ptr<_Tp, _Dp>::unique_ptr(std::nullptr_t) [with _Tp = font::floating_label_context; _Dp = std::default_delete<font::floating_label_context>; std::nullptr_t = std::nullptr_t]’
src/editor/controller/editor_controller.cpp:78:18:   required from here
/usr/include/c++/6/bits/unique_ptr.h:74:22: error: invalid application of ‘sizeof’ to incomplete type ‘font::floating_label_context’
  static_assert(sizeof(_Tp)>0,
                      ^
```

It’s a side-effect of removing the inclusion of `filechooser.hpp` from  `gui/dialogs/editor/custom_tod.hpp`.